src/pages/LoanFinding/LoanFinding.vue

```diff
@@ -181,16 +193,26 @@ export default {
 			enableQFMobileVersion: false,
 			enableAlmostFundedRow: false,
 			HandOrangeIcon,
+			enableLoanRecommendations: false,
 		};
 	},
 	apollo: {
 		preFetch(config, client) {
 			return client.query({
 				query: experimentAssignmentQuery, variables: { id: THREE_LOANS_RECOMMENDED_ROW_EXP_KEY }
-			}).then(() => {
+			}).then(async () => {
+				const loanRecommendationsExp = await client.query({
```

Comment:
Does the await here work as expected? Asking because this `preFetch` logic is a custom implementation that we built, and we don't use async await with `preFetch` anywhere else in `ui`. I know, for example, that async await doesn't work as expected in the created method.

Reply:
One thing to try here is actually removing this `experimentAssignmentQuery` entirely and using `readQuery`. We assign all active experiments globally on `preFetch`, so we might not need to run the assignment here.

Reply:
okay, after fixing the chaining, it still works - so I will push up that version in a bit, but I'm a little confused by the last comment since this IS the prefetch function...?

Reply:
I think he just means using the `result` from the query in another `then` statement so the 2 queries are separated. `exp query` -> `.then(result => recommendationsQuery` with data from the result

Reply:
Ignore the config access...  I just added a userInfo query into the prefetch function.  There's a cache read in the created() function as well which I'm hesitant to remove because I don't actually know how to test user state in dev.  Do either of you know how to do that?

Reply:
Okay I've read this all over a few times but still scratching my head.  @mcstover  are you talking about the promise chaining comment where casey is talking about using .then chaining, or the last comment Casey made about not needing to run an experiment assignment in the prefetch function because we assign all active experiments globally on preFetch?

The part I am confused about is the idea that "we don't need to run assignment query" on prefetch because we are "running an assignment query on prefetch".

Reply:
I actually meant that we run `assignAllActiveExperiments` in `WwwPage`. If I remember right, you still need to run the individual assign queries if you need the assignment in a child `preFetch`, but you also might not, since the parent `preFetch` should run first. The hierarchy of pre fetch calls might run simultaneously though, that's the part I'm forgetting.

Reply:
@dyersituations does this promise.all version look any better?

Reply:
Yup, looks good, queries are now ran in parallel where we can!